### PR TITLE
Feat/nemoclaw local llm support

### PR DESCRIPTION
**Pull request by nvleeryan**
## Summary

Adds NemoClaw-managed local LLM support to the NemoClaw setup notebook (`deploy/docker/scripts/deploy_nemoclaw.ipynb`), makes provider selection deterministic so what you pick in section 1 is what the sandbox is onboarded with, and makes a failed re-onboard non-destructive.

- **Local providers in (b).** Cell (b) now selects a NemoClaw-managed provider (`install-vllm`, `ollama`, `nim-local`, …) rather than only attaching to a server you started yourself, which remains available as `custom`. For `install-vllm`, `NEMOCLAW_MODEL` takes a serving-catalog slug (`nemotron-3.5-lightning-30b`, `muse-glimmer-30b`, …) that 3.1 forwards as `NEMOCLAW_VLLM_MODEL`, and blank takes the catalog default for the host. `HF_TOKEN` is passed through for gated weight downloads, and `NEMOCLAW_EXPERIMENTAL=1` is set for the local providers.
- **vLLM placement.** New `NEMOCLAW_VLLM_PORT` (default `18000`, range-checked) and `NEMOCLAW_VLLM_GPU_DEVICE` (default `"0"`) keep managed vLLM off the GPUs you want for VSS. `--vllm-gpu-device` is passed only for `install-vllm`, since other providers reject it. Before onboarding, 3.1 probes the port the same way NemoClaw does (`/v1/models`): a listener that serves no model fails early with an explanation, and a vLLM that is already serving drops the GPU flag, because NemoClaw attaches to that server and aborts if the flag is present.
- **Provider config can no longer be inherited by accident.** `NEMOCLAW_PROVIDER` is seeded in 1.1 and set explicitly by each of (a)/(b)/(c); inferring it from `NEMOCLAW_ENDPOINT_URL` is now only a fallback for an unset provider. 3.1 validates per provider and drops the variables the chosen provider does not consume (`NEMOCLAW_ENDPOINT_URL`, `COMPATIBLE_API_KEY`, `NEMOCLAW_MODEL`/`NEMOCLAW_VLLM_MODEL`) from both `env` and `os.environ`, so a value left by an earlier cell run cannot reach the installer. A contradictory combination raises with the fix named instead of silently selecting a different provider.
- **Blank `COMPATIBLE_API_KEY` is now valid for a self-hosted endpoint.** 3.1 substitutes the `EMPTY` placeholder when the endpoint resolves to a private address, and still requires a real bearer token for a public one. The preflight check moved accordingly.
- **`NEMOCLAW_RECREATE_SANDBOX` (default `True`).** Provider, model, endpoint, key, `AGENT_RUNTIME`, and vLLM GPU all bind at onboard, and onboard cannot reconfigure a sandbox that already exists — so 3.1 re-onboards with `--recreate-sandbox` by default, and 3.2–3.6 reinstall the policy, skills, and workspace docs. The replacement happens inside onboard's sandbox phase, after preflight, gateway, provider and inference, so a configuration onboard refuses fails with the existing sandbox intact rather than leaving the host with none. `False` reuses the sandbox and skips onboard; the shell override is `1`/`0`, the spelling NemoClaw reads for that same variable name.
- **Overrides read a snapshot, not live env.** Shell overrides now come from `SHELL_ENV`, captured once per kernel, so re-running 1.3 after editing a value picks up the edit rather than what 3.1 exported on the previous run.
- **Observability and pin.** Onboard echoes its resolved environment with secrets masked, prints the recorded provider and model from `~/.nemoclaw/onboard-session.json`, and the last retry adds `--fresh` (carrying `--recreate-sandbox` when replacing a sandbox, since `--fresh` skips resume detection). Installer pinned to `v0.0.114`; 1.3 is now a documented bullet list.

## Behavior change to note

3.1 replaces the sandbox by default: a successful re-run discards an existing sandbox and its agent sessions, and 3.2–3.6 reinstall everything on the new one. A *failed* run does not — onboard replaces the sandbox only after the configuration, gateway, provider, and inference steps pass, so a rejected setting leaves the current sandbox in place. To reuse instead, set `NEMOCLAW_RECREATE_SANDBOX = False`, or `NEMOCLAW_RECREATE_SANDBOX=0` in the launching shell for unattended runs.

## Test plan

- [x] (c) `build` provider end-to-end on a 2× RTX PRO 6000 host: recreate, onboard, policy, skills, workspace docs, UI reachable.
- [x] (b) `install-vllm` with `NEMOCLAW_VLLM_GPU_DEVICE` set, confirming vLLM binds `NEMOCLAW_VLLM_PORT` on that GPU only.
- [x] (b) `install-vllm` with a vLLM already serving on that port: GPU flag is dropped and onboard attaches instead of installing.
- [x] (b) `install-vllm` with a non-model listener on the port: 3.1 fails early with the port message.
- [x] (b) `install-vllm` with an unrecognised `NEMOCLAW_MODEL` slug against an existing sandbox: onboard fails up front and that sandbox is still usable afterwards.
- [x] (a) `custom` against a public endpoint (missing `COMPATIBLE_API_KEY` raises) and a private one (placeholder accepted, host proxy started).
- [x] Run (b) then (c) in one kernel and confirm onboard reports `build` with no endpoint in the echoed environment.
- [x] `NEMOCLAW_RECREATE_SANDBOX = False` reuses an existing sandbox and 3.2–3.6 still pass against it.
- [x] `NEMOCLAW_RECREATE_SANDBOX=0` in the launching shell reuses it too, and `=1` replaces it.
- [x] `NEMOCLAW_INSTALL_REF=… jupyter nbconvert --execute` honors the shell override.

## Checklist
- [ ] I am familiar with the [Contributing Guidelines](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/blob/HEAD/CONTRIBUTING.md).
- [ ] I have installed and run [pre-commit hooks](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/blob/HEAD/CONTRIBUTING.md#1-enable-pre-commit-hooks) locally (`uv run pre-commit install` once, then hooks run on every `git commit`).
- [ ] Every commit on this PR is [DCO sign-off](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/blob/HEAD/CONTRIBUTING.md#developer-certificate-of-origin-dco)'d (`git commit -s` adds a `Signed-off-by` trailer that certifies you have the right to submit the change under Apache-2.0).
- [ ] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.
- [ ] No secrets, API keys, or credentials committed.

### CODE REVIEW

**Comment by nvleeryan** [comment]
/ok to test https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1781/commits/8f591756e8af013445b63416dfad83347472d423

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/32546865860)
- Internal approval evidence was evaluated by the protected compliance pipeline.

File: deploy/docker/scripts/deploy_nemoclaw.ipynb [hunk 1]
```diff
@@ -195,6 +212,11 @@
         "AGENT_HOOKS_ENABLED = True     # enable agent inbound webhooks when the harness supports them\n",
         "AGENT_HOOKS_PATH = \"/hooks\"     # OpenClaw hooks path (ignored for Hermes)\n",
         "AGENT_WEBHOOK_PORT = 8644       # Hermes webhook adapter port (ignored for OpenClaw)\n",
+        "NEMOCLAW_VLLM_PORT = 18000     # Host port for NemoClaw-hosted vLLM\n",
+        "# GPU for NemoClaw-hosted vLLM: an index (\"1\") or a full GPU/MIG UUID.  Blank leaves the provider's default (all\n",
+        "# visible GPUs).\n",
+        "NEMOCLAW_VLLM_GPU_DEVICE = \"0\"     # 0-based index (0 = first GPU)\n",
+        "NEMOCLAW_RECREATE_SANDBOX = True     # Destroy an existing sandbox in 3.1 before onboarding it again. \n",
```
[the remark this answers is not in the document]
Comment (nvleeryan) [reply deploy/docker/scripts/deploy_nemoclaw.ipynb]: For recovery we chose to recreate the sandbox on every 3.1 run (NEMOCLAW_RECREATE_SANDBOX = True, the default) rather than teach the notebook to reconfigure an existing one and handle the long tail of edge cases.  Corresponding comments were updated as well

**Comment by nvleeryan** [comment]
/ok to test https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1781/commits/1d1a6699beba066349f3e582f3301af937566414

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/32917302583)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Comment by nvleeryan** [comment]
/ok to test https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/pull/1781/commits/540376475c07135462daaa36be58964e8441de6e

**Review by zac-wang-nv — COMMENTED** [review]
_This is Hinton, the VSS CI OSRB reviewer_

## Automated OSRB Review — INCONCLUSIVE

The internal review failed closed: required evidence collection had errors

- Dependency changes evaluated: `0`
- Public evidence: [License Diff Actions run](https://github.com/NVIDIA-AI-Blueprints/video-search-and-summarization/actions/runs/32920985277)
- Internal approval evidence was evaluated by the protected compliance pipeline.

**Review by zac-wang-nv — APPROVED** [review]
LGTM, 

TODO: fix the skill eval harness.
